src/Core/Services/Implementations/UserService.cs

```diff
@@ -567,6 +567,28 @@ public override Task<IdentityResult> ChangePasswordAsync(User user, string maste
             Logger.LogWarning("Change password failed for user {userId}.", user.Id);
             return IdentityResult.Failed(_identityErrorDescriber.PasswordMismatch());
         }
+        
+        public async Task<IdentityResult> ChangePasswordNoCompareAsync(User user, string newMasterPassword, string key)
```

Comment:
I would assume we definitely will want a check/validation here that the user does **not** already have a password hash stored, and therefore can set one w/o comparison... @kspearrin , can you confirm this assumption or would we want to leave this more open for when we do emergency access, etc.?

Reply:
Yes, definitely need to check that. See my previous comment. It would probably be here in the service.

Reply:
Is this name for this method okay? Or should it match the `SetPasswordAsync` format from the API controller?

Reply:
I'd recommend renaming too.

Reply:
Updated.